packages/mgt-components/src/components/mgt-people-picker/mgt-people-picker.ts

```diff
@@ -1032,6 +1037,9 @@ export class MgtPeoplePicker extends MgtTemplatedTaskComponent {
         !this.defaultSelectedUsers.length &&
         !this.defaultSelectedGroups.length
       ) {
+        if (this.hasMaxSelections) {
```

Comment:
I'd move this check to be after the async operations complete and `this.selectedPeople` has been updated.